pkg/controller/controller_create_test.go

```diff
@@ -0,0 +1,85 @@
+package controller
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/ablarry/golang-bootcamp-2020/pkg/model"
+	"github.com/ablarry/golang-bootcamp-2020/pkg/repo"
+	"github.com/ablarry/golang-bootcamp-2020/pkg/service"
+	"github.com/ablarry/golang-bootcamp-2020/pkg/utils/errors"
+	"github.com/gorilla/mux"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestCreate test Create mocking service in controller
+func TestCreate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	table := []struct {
+		name string
+		*http.Request
+		code        int
+		mockService func(c *gomock.Controller) *service.MockService
+		params      map[string]string
+		expected    *model.Pokemon
+		ok          bool
+	}{
```

Comment:
For this and any other struct with no named fields.

I'd rather have all fields named, and also in the test cases have them named for better readability and understanding of the code.

```suggestion
	table := []struct {
		name        string
		req         *http.Request
		code        int
		mockService func(c *gomock.Controller) *service.MockService
		params      map[string]string
		expected    *model.Pokemon
		ok          bool
	}{
```